packages/vite/index.cjs

```diff
@@ -32,3 +34,14 @@ unsupportedCJS.forEach((name) => {
     )
   }
 })
+
+function warnCjsUsage() {
+  if (process.env.VITE_CJS_IGNORE_WARNING) return
+  const yellow = (str) => `\u001b[33m${str}\u001b[39m`
+  const log = process.env.VITE_CJS_TRACE ? console.trace : console.warn
+  log(
+    yellow(
+      `The CJS build of Vite's Node API is deprecated. See https://vitejs.dev/guide/troubleshooting.html#vite-cjs-node-api-deprecated for more details.`,
+    ),
+  )
```

Comment:
The warning gets hidden because Vite moves the position of the terminal (https://github.com/vitejs/vite/issues/9378).
<video src="https://github.com/vitejs/vite/assets/49056869/cdc4cb09-c441-4112-bd99-9d7c7704cc02">

I'm not sure how we can fix this though. 🤔

Reply:
Yeah I noticed it too. We could hack around and fix it if we're running the `vite [...]` command, but I think probably best to leave it as is for now until we have a solution for #9378

Reply:
How about buffering this via a global flag until the logger is created?
Then we can log this warning using the logger, which already has the "avoid clearing the screen when there is a warning" behavior.

Reply:
It's still tricky though since we want to log for all APIs, and not just dev or build only. For example, if someone is only using Vite via `const { normalizePath } = require('vite')`, then it could never get logged. Plus it makes the tracing message harder to implement too.

Maybe we can keep using the global flag approach as a boolean, then use that to judge whether to clear the screen on the initial startup or not.

Reply:
ce7b03b should fix it by not clearing the screen if the cjs warning is logged.